pkg/deploy/reaper/reaper_test.go

```diff
@@ -103,36 +95,86 @@ func TestStop(t *testing.T) {
 				"delete-replicationController",
 			},
 			output: "config stopped",
+			err:    false,
+		},
+		{
+			testName:  "no config, some deployments",
+			namespace: "default",
+			name:      "config",
+			osc:       testclient.NewSimpleFake(notfound()),
+			kc:        ktestclient.NewSimpleFake(mkdeploymentlist(1)),
+			expected: []string{
+				"delete-deploymentconfig",
+			},
+			kexpected: []string{
+				"list-replicationControllers",
+				"get-replicationController",
+				"update-replicationController",
+				"get-replicationController",
+				"delete-replicationController",
+			},
+			output: "config stopped",
+			err:    false,
+		},
+		{
+			testName:  "no config, no deployments",
+			namespace: "default",
+			name:      "config",
+			osc:       testclient.NewSimpleFake(notfound()),
+			kc:        ktestclient.NewSimpleFake(&kapi.ReplicationControllerList{}),
+			expected: []string{
+				"delete-deploymentconfig",
+			},
+			kexpected: []string{
+				"list-replicationControllers",
+			},
+			output: "",
+			err:    true,
+		},
+		{
+			testName:  "config, no deployments",
+			namespace: "default",
+			name:      "config",
+			osc:       testclient.NewSimpleFake(deploytest.OkDeploymentConfig(5)),
+			kc:        ktestclient.NewSimpleFake(&kapi.ReplicationControllerList{}),
+			expected: []string{
+				"delete-deploymentconfig",
+			},
+			kexpected: []string{
+				"list-replicationControllers",
+			},
+			output: "config stopped",
+			err:    false,
 		},
 	}
 
-	if len(fakeClients) != len(tests) {
-		t.Fatalf("no. of clients should equal to the no. of tests. Fix those tests.")
-	}
-
-	for i, test := range tests {
-		out, err := test.reaper.Stop(test.namespace, test.name, nil)
-		if err != nil {
-			t.Fatalf("%s: unexpected error: %v", test.testName, err)
+	for _, test := range tests {
+		reaper := &DeploymentConfigReaper{osc: test.osc, kc: test.kc, pollInterval: time.Millisecond, timeout: time.Millisecond}
+		out, err := reaper.Stop(test.namespace, test.name, nil)
+		if !test.err && err != nil {
+			t.Errorf("%s: unexpected error: %v", test.testName, err)
+		}
+		if test.err && err == nil {
+			t.Errorf("%s: expected an error", test.testName)
 		}
-		if len(fakeClients[i].Actions) != len(test.expected) {
-			t.Fatalf("%s: unexpected actions: %v, expected %v", test.testName, fakeClients[i].Actions, test.expected)
+		if len(test.osc.Actions) != len(test.expected) {
+			t.Errorf("%s: unexpected actions: %v, expected %v", test.testName, test.osc.Actions, test.expected)
```

Comment:
This should stay Fatal otherwise if the length of the slices is not equal we'll get a panic in the following stanza:)